SalesforceCommerceCloud/pwa-kit
# Show warning when using incompatible node version

# Description

Ticket: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000fOo0rYAC/view

This PR makes sure to show a warning when a developer uses an incompatible node version while doing some actions (`npm ci`/`npm install`, `npm start`, generating a new project). The PR does 2 main things:
- Making sure that all packages have defined and use consistent `engines` field in their package.json
- Within our generator script, perform a check for the developers' node version and show a warning if it's incompatible

This PR builds upon what we already have.. there's already an existing mechanism that does this kind of check. For example, before this PR, run `npm ci` at the root of the monorepo and you'll encounter our script that asks whether you want to proceed or not with your incompatible node. But we found that the check is not done in some areas like the generator.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# How to Test-Drive This PR

First, try testing the generator:
1. Switch to the root folder
2. Use an incompatible Node version: `nvm use 13`
3. Generate a project with `node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir /tmp/project`
4. You should see a warning like this
![NeoVim Terminal 2022-02-08 at 17 33 09](https://user-images.githubusercontent.com/847300/153107659-159b9d10-a6b8-4a04-ac66-3bcfdb1ca220.png)

Then with your generated project:
1. `cd /tmp/project`
2. Use an incompatible node: `nvm use 16` ← I found version 13 to not show a warning message
2. `npm ci`
3. You should see a warning like this. It's the default message that comes from npm.
![NeoVim Terminal 2022-02-08 at 17 31 08](https://user-images.githubusercontent.com/847300/153107799-56b53072-1691-4fae-848b-b6119a9b161a.png)

Also, try running the dev server with your project:
1. `npm start`
2. You should see a warning like this. FYI I didn't have to implement this, as it already exists.
![NeoVim Terminal 2022-02-08 at 17 32 05](https://user-images.githubusercontent.com/847300/153107899-bc8b3ea5-ba17-4527-b995-b76aa87be877.png)


# Checklists




## General



## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
